packages/token-bridge-sdk/src/util/index.ts

```diff
@@ -14,3 +23,107 @@ export const validateTokenList = (tokenList: TokenList) => {
 
   return validate(tokenList)
 }
+
+/**
+ * Retrieves data about an ERC-20 token using its L1 address. Throws if fails to retrieve balance or allowance.
+ * @param erc20L1Address
+ * @returns
+ */
+export async function getL1TokenData(
+  erc20L1Address: string,
+  params: TokenBridgeParams,
+  throwOnInvalidERC20 = true
+): Promise<L1TokenData> {
```

Comment:
Let's try to decouple these functions as much as possible from how the `useArbTokenBridge` hook is structured. I think we can make the signature look like this:

```suggestion
export async function getL1TokenData({
  account,
  erc20L1Address,
  l1Provider,
  l2Provider,
  throwOnInvalidERC20 = true
}: {
  account: string
  erc20L1Address: string
  l1Provider: Provider
  l2Provider: Provider
  throwOnInvalidERC20?: boolean
}): Promise<L1TokenData> {
```

We can instantiate the `l2Network` like this: `getL2Network(l2Provider)`. Let's do the same with `getL2TokenData`